src/timer.rs

```diff
@@ -58,35 +32,34 @@ mod test {
     use std::time::Instant;
 
     use log::debug;
-    use tokio::select;
-
     use fluvio_future::test_async;
     use fluvio_future::timer::sleep;
 
+    const SLEEP_MS: u64 = 10;
+    const ITERATIONS: u16 = 10;
+
     /// test timer loop
     #[test_async]
     async fn test_sleep() -> Result<(), ()> {
-        let mut sleep_count: u16 = 0;
         let time_now = Instant::now();
+        let sleep_duration = Duration::from_millis(SLEEP_MS);
 
-        let mut sleep_ft = sleep(Duration::from_millis(10));
+        let mut sleep_count = 0;
 
-        for _ in 0u16..10u16 {
-            select! {
-                _ = &mut sleep_ft => {
-                    // fire everytime but won't make cause more delay than initial 10 ms
-                    sleep_count += 1;
-                    debug!("timer fired");
-                }
-            }
+        while sleep_count < ITERATIONS {
+            sleep(sleep_duration).await;
+            sleep_count +=1 ;
+            debug!("timer fired, sleep_count: {}", sleep_count);
         }
```

Comment:
this is incorrect. this will sleep for SLEEP_MS*iteration instead of just a single SLEEP_MS